docs/module/gateway.md

```diff
@@ -0,0 +1,63 @@
+
+The module supports statically configured anycast gateway IPv4 address and VRRPv3 for IPv4 and IPv6.
+
+The module is supported on these platforms:
+
+| Operating system      | Anycast | VRRPv3 | VRRPv3<br>IPv6 |
+| --------------------- | :-: | :-: | :-: |
+| Arista EOS            | ✅  | ✅  | ✅  |
+| Cisco CSR 1000v       | ✅  | ✅  | ✅  |
+| Cisco IOSv            | ✅  | ✅  | ✅  |
+| Cumulus Linux         | ✅  | ✅  | ✅  |
+
+## Global Parameters
+
+The module supports the following global parameters: 
+
+* **gateway.protocol** (default: *anycast*) -- the first-hop gateway resolution protocol. The only supported value is currently *anycast*
+* **gateway.id** (default: -1) -- the IP address within the subnet used for the gateway IP address
+
+## Link Parameters
+
+Gateway configuration module is enabled on all links that have **gateway** attribute set to *True* or to a dictionary of valid parameters. You can change most global parameters on per-link basis.
+
+## Anycast Gateway
+
+The *gateway* configuration module supports IPv4 anycast gateways -- MAC and IPv4 addresses shared between multiple nodes attached to the same segment.
+
+### Implementation Notes
+
+* The default *netlab* shared MAC address is 0200.cafe.00ff.
+* Vendors use different names for anycast gateways: VARP (Arista), VRR (Cumulus)
+* Arista EOS uses the same shared MAC address on all interfaces. Do not set link-level **gateway.anycast.mac** parameter in topologies using Arista EOS.
+* Arista vEOS cannot use VRRP MAC address as a shared MAC address (default recommended by Cumulus Linux)
+
+### Anycast Parameters
+
+Anycast implementation of shared first-hop IPv4 address supports these parameters that can be specified globally or on individual links or interfaces.
+
+* **gateway.anycast.unicast** (default: True) -- configure node-specific unicast IP addresses together with anycast IP address.
+* **gateway.anycast.mac** -- Static MAC address used for the anycast IP address
+
+```{tip}
+Many implementations require unique unicast IP addresses configured on the interfaces that have anycast IP address. Set **gateway.anycast.unicast** to *False* only when absolutely necessary.
+```
+
+## Virtual Router Redundancy Protocol (VRRP)
+
+*netlab* supports a single VRRPv3 instance per subnet. The VRRPv3 instance can provide shared IPv4 and IPv6 addresses. More complex topologies like multiple VRRPv3 instances could be deployed with a judicious application of interface parameters[^VNS].
+
+[^VNS]: These topologies are not supported and will not be integrated into *netlab* core. If you want to have an easier way of configuring the in a lab topology, please feel free to create a plugin.
+
+### VRRP Parameters
+
+*netlab* supports these VRRP parameters that can be specified globally or on individual links or interfaces.
+
+* **vrrp.group** (default: 1) -- VRRP group
```

Comment:
On SR OS, vrrp.group influences the virtual MAC address that gets generated. In a LAN segment with multiple pairs of VRRP gateways, one would have to ensure unique group ids - is that something we want to check/enforce?